flepimop/gempyor_pkg/src/gempyor/utils.py

```diff
@@ -346,3 +353,164 @@ def bash(command):
+    """
+    resume_discard_seeding = os.environ.get("RESUME_DISCARD_SEEDING")
+    flepi_block_index = os.environ.get("FLEPI_BLOCK_INDEX")
+    if flepi_block_index == "1":
+        if resume_discard_seeding == "true":
+            return ["spar", "snpi", "hpar", "hnpi", "init"]
+        else:
+            return ["seed", "spar", "snpi", "hpar", "hnpi", "init"]
+    else:
+        return ["seed", "spar", "snpi", "hpar", "hnpi", "host", "llik", "init"]
+
+
+def create_resume_file_names_map() -> Dict[str, str]:
+    """
+    Generates a mapping of input file names to output file names for a resume process based on
+    parquet file types and environmental conditions. The function adjusts the file name mappings
+    based on the operational block index and the location of the last job output.
+
+    The mappings depend on:
+    - Parquet file types appropriate for resuming a process, as determined by the environment.
+    - Whether the files are for 'global' or 'chimeric' types, as these liketypes influence the
+      file naming convention.
+    - The operational block index ('FLEPI_BLOCK_INDEX'), which can alter the input file names for
+      block index '1'.
+    - The presence and value of 'LAST_JOB_OUTPUT' environment variable, which if set to an S3 path,
+      adjusts the keys in the mapping to be prefixed with this path.
+
+    Returns:
+        Dict[str, str]: A dictionary where keys are input file paths and values are corresponding
+                        output file paths. The paths may be modified by the 'LAST_JOB_OUTPUT' if it
+                        is set and points to an S3 location.
+
+    Raises:
+        No explicit exceptions are raised within the function, but it relies heavily on external
+        functions and environment variables which if improperly configured could lead to unexpected
+        behavior.
+    """
+    parquet_types = get_parquet_types_for_resume()
+    resume_file_name_mapping = dict()
+    liketypes = ["global", "chimeric"]
+    for filetype in parquet_types:
+        for liketype in liketypes:
+            output_file_name = create_resume_out_filename(filetype=filetype, liketype=liketype)
+            input_file_name = output_file_name
+            if os.environ.get("FLEPI_BLOCK_INDEX") == "1":
+                input_file_name = create_resume_input_filename(filetype=filetype, liketype=liketype)
+            resume_file_name_mapping[input_file_name] = output_file_name
+
+    last_job_output = os.environ.get("LAST_JOB_OUTPUT")
+    if last_job_output.find("s3://") >= 0:
+        old_keys = list(resume_file_name_mapping.keys())
+        for k in old_keys:
+            new_key = os.path.join(last_job_output, k)
+            resume_file_name_mapping[new_key] = resume_file_name_mapping[k]
+            del resume_file_name_mapping[k]
+    return resume_file_name_mapping
+
+
+def download_file_from_s3(name_map: Dict[str, str]) -> None:
```

Comment:
Amazing this is very good. Ideally we also have a "move on filesystem" function that can be used as drop-in when resuming from another folder. In this case, it should just move the file in a safe way (creating folders along the way)